src/blocks/pomodoro.rs

```diff
@@ -0,0 +1,212 @@
+use crossbeam_channel::Sender;
+use std::process::Command;
+use std::time::Duration;
+
+use crate::blocks::{Block, ConfigBlock};
+use crate::config::Config;
+use crate::errors::*;
+use crate::input::{I3BarEvent, MouseButton};
+use crate::scheduler::Task;
+use crate::widget::I3BarWidget;
+use crate::widgets::text::TextWidget;
+
+use uuid::Uuid;
+
+enum State {
+    Started,
+    Stopped,
+    Paused,
+    OnBreak,
+}
+
+pub struct Pomodoro {
+    id: String,
+    time: TextWidget,
+    state: State,
+    elapsed: usize,
+    length: usize,
+    break_length: usize,
+    update_interval: Duration,
+    message: String,
+    break_message: String,
+    count: usize,
+}
+
+#[derive(Deserialize, Debug, Default, Clone)]
+#[serde(deny_unknown_fields)]
+pub struct PomodoroConfig {
+    #[serde(default = "PomodoroConfig::default_length")]
+    pub length: usize,
+    #[serde(default = "PomodoroConfig::default_break_length")]
+    pub break_length: usize,
+    #[serde(default = "PomodoroConfig::default_message")]
+    pub message: String,
+    #[serde(default = "PomodoroConfig::default_break_message")]
+    pub break_message: String,
+}
+
+impl PomodoroConfig {
+    fn default_length() -> usize {
+        25
+    }
+
+    fn default_break_length() -> usize {
+        5
+    }
+
+    fn default_message() -> String {
+        "Pomodoro over! Take a break!".to_owned()
+    }
+
+    fn default_break_message() -> String {
+        "Break over! Time to work!".to_owned()
+    }
+}
+
+impl ConfigBlock for Pomodoro {
+    type Config = PomodoroConfig;
+
+    fn new(block_config: Self::Config, config: Config, _send: Sender<Task>) -> Result<Self> {
+        let id: String = Uuid::new_v4().simple().to_string();
+        let id_copy = id.clone();
+
+        Ok(Pomodoro {
+            id: id_copy,
+            time: TextWidget::new(config).with_icon("pomodoro"),
+            state: State::Stopped,
+            length: block_config.length * 60,             // convert to minutes
+            break_length: block_config.break_length * 60, // convert to minutes
+            update_interval: Duration::from_millis(1000),
+            message: block_config.message,
+            break_message: block_config.break_message,
+            elapsed: 0,
+            count: 0,
+        })
+    }
+}
+
+impl Block for Pomodoro {
+    fn id(&self) -> &str {
+        &self.id
+    }
+
+    fn update(&mut self) -> Result<Option<Duration>> {
+        self.tick();
+        self.set_text();
+
+        match &self.state {
+            State::Started => {
+                if self.elapsed >= self.length {
+                    let message = self.message.to_owned();
```

Comment:
there must be a better way to do this that avoids using `move` and instead uses lifetimes, but this seemed ok?

Reply:
Once you remove the thread spawn this should resolve itself; you can just borrow `self.message`.

Reply:
but I will need `thread.spawn` to be able to use an external command and don't wait for it to return whether it's nagbar or something else 

am I missing something?